packages/bot-web-ui/src/components/dashboard/dashboard-component/load-bot-preview/recent-workspace.tsx

```diff
@@ -0,0 +1,119 @@
+import classnames from 'classnames';
+import React from 'react';
+import { Icon } from '@deriv/components';
+import { timeSince } from '@deriv/bot-skeleton';
+import { save_types } from '@deriv/bot-skeleton/src/constants/save-type';
+import { connect } from 'Stores/connect';
+import RootStore from 'Stores/index';
+import './index.scss';
+
+type TRecentWorkspace = {
+    getRecentFileIcon: (string: string) => void;
+    getSaveType: () => void;
+    previewRecentStrategy: () => void;
+    selected_strategy_id: string;
+    workspace: { [key: string]: string };
+    setFileLoaded: (param: boolean) => void;
+    index: number;
+    has_file_loaded: boolean;
+    toggleSaveModal: () => void;
+    toggleDeleteDialog: () => void;
+    loadFileFromRecent: () => void;
+    setActiveTab: (active_tab: number) => void;
+};
+
+const RecentWorkspace = ({
+    getRecentFileIcon,
+    getSaveType,
+    previewRecentStrategy,
+    selected_strategy_id,
+    workspace,
+    index,
+    has_file_loaded,
+    setFileLoaded,
+    toggleDeleteDialog,
+    toggleSaveModal,
+    loadFileFromRecent,
+    setActiveTab,
+}: TRecentWorkspace) => {
+    const trigger_div_ref = React.useRef<HTMLInputElement | null>(null);
+    React.useEffect(() => {
+        if (index === 0 && has_file_loaded === false) {
+            trigger_div_ref?.current?.click();
```

Comment:
any reason to use ref in this case?

Reply:
yes we used this to select the first strategy when the compoent renders